security/yubi-key.md

```diff
@@ -40,52 +40,67 @@ password (other than your login password!) and apply the configuration.
+   rebooting your USB VM.
+
+2. Configure your YubiKey for challenge-reponse `HMAC-SHA1` mode, for example
+   [following this
+   tutorial](https://www.yubico.com/products/services-software/personalization-tools/challenge-response/).
 
-       # if password has given, verify it
-       if [ -n "$2" ]; then
-           # PAM appends \0 at the end
-           hash=`head -c -1 | openssl dgst -sha1 -r | cut -f1 -d ' '`
-           if [ "x$2" != "x$hash" ]; then
-               exit 1
-           fi
-       fi
+   On Debian, you can run the graphical user interface `yubikey-personalization-gui` from the command line.
 
-       challenge=`head -c64 /dev/urandom | xxd -c 64 -ps`
-       # You may need to adjust slot number and USB VM name here
-       response=`qvm-run -u root --nogui -p sys-usb "ykchalresp -2 -x $challenge"`
+   - Choose `configuration slot 2`.
+   - It is recommended to enable `Require user input (button press)` but this is optional.
+   - Note: Derivating from the above video, use the following settings select
+   `HMAC-SHA1 mode`: `fixed 64 bit input`.
+   - We will refer the `Secret Key (20 bytes hex)` as `AESKEY`.
+      - It is recommended to keep a backup of your `AESKEY` in an offline VM used as vault.
+      - Consider to keep a backup of your `AESKEY` on paper and store it in a safe place.
+      - In case you have multiple yubikeys for backup purposes (in case a yubikey gets lost, stolen or breaks) you can write the same settings into other yubikeys.
 
-       correct_response=`echo $challenge | xxd -r -ps | openssl dgst -sha1 -macopt hexkey:$key -mac HMAC -r | cut -f1 -d ' '`
+3. Install [qubes-app-yubikey](https://github.com/QubesOS/qubes-app-yubikey) in dom0.
 
-       test "x$correct_response" = "x$response"
-       exit $?
+       sudo qubes-dom0-update qubes-yubikey-dom0
 
-4. Adjust USB VM name (`sys-usb` above), and possibly YubiKey slot number (`2`
-   above), then make the script executable.
-5. Edit `/etc/pam.d/xscreensaver` (or appropriate file if you are using other
-   screen locker program). Add this line at the beginning:
+4. Adjust USB VM name in case you are using something other than the default
+   `sys-usb` by editing `/etc/qubes/yk-keys/yk-vm` in dom0.
 
-       auth [success=done default=ignore] pam_exec.so expose_authtok quiet /usr/local/bin/yubikey-auth AESKEY PASSWORD-HASH
+5. Paste your `AESKEY` from step 2 into `/etc/qubes/yk-keys/yk-secret-key.hex` in dom0.
 
-   Replace `AESKEY` with hex-encoded key configured in the first step, then
-   replace `PASSWORD-HASH` with SHA1 hash for your YubiKey-linked password (other
-   than your standard Qubes password). You can calculate it using this command:
+6. Paste your hashed password (other than your standard Qubes password)  into
+`/etc/qubes/yk-keys/yk-login-pass-hashed.hex` in dom0.
+
+   You can calculate your hashed password using this command:
 
        echo -n "PASSWORD" | openssl dgst -sha1
 
+   (Replace `PASSWORD` with your actual password.)
```

Comment:
Technically this is ok, but it will leave the password in shell history. I recommend this instead:
```
read password
echo -n "$password" | openssl dgst -sha1
```